multicluster/service-mirror/cluster_watcher.go

```diff
@@ -1012,22 +1000,51 @@ func (rcsw *RemoteClusterServiceWatcher) repairEndpoints(ctx context.Context) er
 	return nil
 }
 
-func (rcsw *RemoteClusterServiceWatcher) createOrUpdateEndpoints(ctx context.Context, ep *corev1.Endpoints) error {
-	_, err := rcsw.localAPIClient.Client.CoreV1().Endpoints(ep.Namespace).Get(ctx, ep.Name, metav1.GetOptions{})
+// createOrUpdateGatewayEndpoints will create or update the gateway mirror
+// endpoints for a remote cluster. These endpoints are required for the probe worker
+// responsible for probing gateway liveness.
+func (rcsw *RemoteClusterServiceWatcher) createOrUpdateGatewayEndpoints(ctx context.Context, addressses []v1.EndpointAddress) error {
```

Comment:
these gateway mirrors get special treatment in that we never set them to not ready because they're needed by the probes themselves.  we should add some comments explaining this, because it's pretty confusing.

Reply:
Hm I tried to explain that with this existing comment. I can add to it.